src/app/portlets/content-types/fields/content-type-fields-variables/content-type-fields-variables.component.ts

```diff
@@ -96,36 +124,63 @@ export class ContentTypeFieldsVariablesComponent implements OnInit, OnChanges, O
             fieldId: this.field.fieldId,
             variable: variable
         };
-        this.fieldVariablesService.save(params).pipe(take(1)).subscribe(
-            (savedVariable: FieldVariable) => {
-                if (typeof variableIndex !== 'undefined') {
-                    this.fieldVariables = this.updateVariableCollection(
-                        savedVariable,
-                        variableIndex
-                    );
-                } else {
-                    this.fieldVariables = [].concat(savedVariable, this.fieldVariables);
-                    this.canSaveFields = [].concat(null, this.canSaveFields);
+        if (typeof variableIndex !== 'undefined') {
+            this.fieldVariablesService.save(params).pipe(take(1)).subscribe(
+                (savedVariable: FieldVariable) => {
+                    if (typeof variableIndex !== 'undefined') {
+                        this.fieldVariables = this.updateVariableCollection(
+                            savedVariable,
+                            variableIndex
+                        );
+                    } else {
+                        this.fieldVariables = [].concat(savedVariable, this.fieldVariables);
+                    }
+                },
+                (err: ResponseView) => {
+                    this.dotHttpErrorManagerService.handle(err).pipe(take(1)).subscribe();
                 }
-            },
-            (err: ResponseView) => {
-                this.dotHttpErrorManagerService.handle(err).pipe(take(1)).subscribe();
-            }
-        );
+            );
+        } else {
+            const emptyVariable: FieldVariable = {
+                key: '',
+                value: ''
+            };
+            this.fieldVariables = [].concat(emptyVariable, this.fieldVariables);
+        }
     }
 
     /**
-     * Holds status if a existing variable meets conditions to be updated
-     * @param {any} event
+     * Handle Add new variable button event
      * @memberof ContentTypeFieldsVariablesComponent
      */
-    updateSaveButtons(event: {
-        column: Column;
-        data: { [key: string]: string };
-        index: number;
-    }): void {
-        this.canSaveFields[event.index] =
-            event.data.key.length > 0 && event.data.value.length > 0 ? null : true;
+    addVariable(): void {
+        this.saveVariable(null);
+        setTimeout(() => {
```

Comment:
Why?

Reply:
when a new variable is going to be added, we need to `timeOut` to let the UI update and then we can reach the new empty row in the UI, by `clicking` it